programs/github

```diff
@@ -21,13 +21,48 @@ function progressbar(f)
 	return left .. right
 end
 
+-- get flag arguments
+local acceptedFlags = {
+	'-b' = 'branch',
+	'--branch' = 'branch',
+	'-t' = 'tag',
+	'--tag' = 'tag',
+	'-a' = 'auth',
+	'--auth' = 'auth'
+}
+
+local findFlag = function(arg) return acceptedFlags[arg] end
+
+local passed = {...}
+local args = {}
+local flags = {}
+local flagValue = ''
+for k, arg in pairs(passed) do
+	local flagName = findFlag(arg)
+	if flagName and arg ~= flagValue then
+		flags[flagName] = passed[k + 1]
+		flagValue = flags[flagName]
+	elseif arg ~= action and arg ~= flagValue then
+		table.insert(args, arg)
+	end
+end
+
+if flags.branch and flags.tag then
+	return printError('please specify a branch or tag.')
```

Comment:
Yeah and honestly providing the tag argument is more of a formality anyway.
Either branch or tag are going to be used for treeName so it's really just
calling the same variable a different name.

On Thu, Aug 25, 2016, 17:42 Eric Wieser <notifications@github.com> wrote:

> In programs/github
> <https://github.com/eric-wieser/computercraft-github/pull/11#discussion_r76338340>
> :
>
> > +local passed = {...}
> > +local args = {}
> > +local flags = {}
> > +local flagValue = ''
> > +for k, arg in pairs(passed) do
> > +	local flagName = findFlag(arg)
> > +	if flagName and arg ~= flagValue then
> > +		flags[flagName] = passed[k + 1]
> > +		flagValue = flags[flagName]
> > +	elseif arg ~= action and arg ~= flagValue then
> > +		table.insert(args, arg)
> > +	end
> > +end
> > +
> > +if flags.branch and flags.tag then
> > +	return printError('please specify a branch or tag.')
>
> Yeah, this conveys the wrong message. "--branch and --tag cannot both be
> specified" would be more accurate.
>
> Note that git clone doesn't actually "support" tags, but git clone -b
> a_tag tends to do the right thing anyway
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eric-wieser/computercraft-github/pull/11/files/0a9cc032fea14dc4adacf2e0d7a51516dbda88e9#r76338340>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA41FnctuDfo_3j-arTEwoTrNKlhxfHfks5qjho4gaJpZM4JsvBA>
> .
>